modules/javafx.base/src/main/java/com/sun/javafx/binding/BidirectionalBinding.java

```diff
@@ -543,15 +574,16 @@ public void changed(ObservableValue<? extends Number> sourceProperty, Number old
         }
     }
 
-    private static class TypedGenericBidirectionalBinding<T> extends BidirectionalBinding<T> {
+    private static class TypedGenericBidirectionalBinding<T> extends BidirectionalBinding {
         private final WeakReference<Property<T>> propertyRef1;
         private final WeakReference<Property<T>> propertyRef2;
+        private T oldValue;
         private boolean updating = false;
 
         private TypedGenericBidirectionalBinding(Property<T> property1, Property<T> property2) {
             super(property1, property2);
-            propertyRef1 = new WeakReference<Property<T>>(property1);
-            propertyRef2 = new WeakReference<Property<T>>(property2);
+            propertyRef1 = new WeakReference<>(property1);
```

Comment:
Don't you need to initialize `oldValue` here?

```
    oldValue = property1.get();
```

Reply:
Yes, it should be initialized.